src/ring_theory/localization.lean

```diff
@@ -1153,6 +1153,152 @@ def at_one [is_localization.away (1 : R) S] : R ≃ₐ[R] S :=
 
 end at_units
 
+section localization_localization
+
+variable (M)
+
+variables (N : submonoid S) (T : Type*) [comm_ring T] [algebra S T]
+variables [algebra R T] [is_scalar_tower R S T]
+
+/--
+Localizing wrt `M ⊆ R` and then wrt `N ⊆ S = M⁻¹R` is equal to the localization of `R` wrt this
+module. See `localization_localization_is_localization`.
+-/
+-- This should only be defined when `S` is the localization `M⁻¹R`, hence the nolint.
+@[nolint unused_arguments]
+def localization_localization_submodule : submonoid R :=
+(N ⊔ M.map (algebra_map R S)).comap (algebra_map R S)
+
+@[simp]
+lemma mem_localization_localization_submodule (x : R) :
+  x ∈ localization_localization_submodule M N ↔
+    ∃ (y : N) (z : M), algebra_map R S x = y * algebra_map R S z :=
+begin
+  rw [localization_localization_submodule, submonoid.mem_comap, submonoid.mem_sup],
+  split,
+  { rintros ⟨y,hy,_,⟨z,hz,rfl⟩,e⟩, exact ⟨⟨y,hy⟩,⟨z,hz⟩,e.symm⟩ },
+  { rintros ⟨y,z,e⟩, exact ⟨y,y.prop,_,⟨z,z.prop,rfl⟩,e.symm⟩ }
+end
+
+lemma localization_localization_map_units [is_localization N T]
+  (y : localization_localization_submodule M N) : is_unit (algebra_map R T y) :=
+begin
+  rcases y with ⟨y, h⟩,
+  rw mem_localization_localization_submodule at h,
+  rcases h with ⟨y, z, eq⟩,
+  rw [is_scalar_tower.algebra_map_eq R S T, ring_hom.comp_apply, subtype.coe_mk, eq,
+      ring_hom.map_mul, is_unit.mul_iff],
+  refine ⟨is_localization.map_units T y,
+    (is_localization.map_units _ z).map (algebra_map S T : S →* T)⟩,
+end
+
+lemma localization_localization_surj [is_localization N T] (x : T) :
+  ∃ (y : R × localization_localization_submodule M N),
+    x * (algebra_map R T y.2) = algebra_map R T y.1 :=
+begin
+  rcases is_localization.surj N x with ⟨⟨y, s⟩, eq₁⟩, -- x = y / s
+  rcases is_localization.surj M y with ⟨⟨z, t⟩, eq₂⟩, -- y = z / t
+  rcases is_localization.surj M s.1 with ⟨⟨z', t'⟩, eq₃⟩, -- s = z' / t'
+  dsimp only at eq₁ eq₂ eq₃,
+  use z * t', use z' * t, -- x = y / s = (z * t') / (z' * t)
+  { rw mem_localization_localization_submodule,
+    refine ⟨s, t * t', _⟩,
+    rw [ring_hom.map_mul, ← eq₃, mul_assoc, ← ring_hom.map_mul, mul_comm t],
+    refl },
+  { simp only [subtype.coe_mk, ring_hom.map_mul, is_scalar_tower.algebra_map_apply R S T,
+      ← eq₃, ← eq₂, ← eq₁],
+    ring! },
+end
```

Comment:
Some more tricks to avoid `refl`/`ring!`:
```suggestion
begin
  rcases is_localization.surj N x with ⟨⟨y, s⟩, eq₁⟩, -- x = y / s
  rcases is_localization.surj M y with ⟨⟨z, t⟩, eq₂⟩, -- y = z / t
  rcases is_localization.surj M (s : S) with ⟨⟨z', t'⟩, eq₃⟩, -- s = z' / t'
  dsimp only at eq₁ eq₂ eq₃,
  use z * t', use z' * t, -- x = y / s = (z * t') / (z' * t)
  { rw mem_localization_localization_submodule,
    refine ⟨s, t * t', _⟩,
    rw [ring_hom.map_mul, ← eq₃, mul_assoc, ← ring_hom.map_mul, mul_comm t, submonoid.coe_mul] },
  { simp only [subtype.coe_mk, ring_hom.map_mul, is_scalar_tower.algebra_map_apply R S T,
      ← eq₃, ← eq₂, ← eq₁],
    ring },
end